### PR TITLE
refactor: Unify error return code

### LINKED ISSUE #37: Unify error return code in TA-endpoint
We used `http_retcode_t` in TA-endpoint to report errors. For more general error coverage, we should rename `http_retcode_t` to `retcode_t`. 
Adding new error status code is necessary, too.

### PR DESCRIPTION
**Pull request by splasky**
These commits fixed the following things:
Renmae `http_retcode_t` to `retcode_t` for general error and status coverage.
Add new status code inside `utils/defined_error.h` if you needed.

closes #37

### CODE REVIEW

**Comment by HowJMay** [comment]
move the change in `connectivity/conn_http.c` and `connectivity/conn_http.h` in #30 to here

**Review by howjmay — CHANGES_REQUESTED** [review]
move the change in connectivity/conn_http.c and connectivity/conn_http.h in #30 to here

File: main.c [hunk 1]
```diff
@@ -40,10 +40,11 @@ void gen_trytes(uint16_t len, char *out) {
   }
 }
 
-void send_https_msg(char const *const host, char const *const port, char const *const api, char const *const tryte_msg,
-                    char const *const addr) {
+int send_https_msg(char const *const host, char const *const port, char const *const api, char const *const tryte_msg,
```
Comment (howjmay) [inline main.c]: Can this return type be changed to `retcode_t`?

**Review by howjmay — CHANGES_REQUESTED** [review]

File: main.c [hunk 2]
```diff
@@ -40,10 +40,11 @@ void gen_trytes(uint16_t len, char *out) {
...
+                   char const *const addr) {
   char req_body[1024] = {}, res[4096] = {0};
   char *req = NULL;
+  int ret = 0;
```
Comment (howjmay) [inline main.c]: ditto

**Review by howjmay — CHANGES_REQUESTED** [review]

File: main.c [hunk 3]
```diff
@@ -55,19 +56,21 @@ void send_https_msg(char const *const host, char const *const port, char const *
   settings.on_body = parser_body_callback;
   http_parser *parser = malloc(sizeof(http_parser));
 
-  while (parser->status_code != HTTP_OK) {
```
Comment (jserv) [inline main.c]: Why did you remove `while`?

**Comment by jserv** [comment]
The changes did not reflect the subject of this pull request.

File: main.c (hunk 1 above)
Comment (splasky) [reply main.c]: Inifinate blocking seem not a big problem for these commits, I would revert it.

**Comment by jserv** [comment]
Don't mix copyright date change and return type tweaks in single pull request.

File: utils/defined_error.h [hunk 4]
```diff
@@ -23,7 +23,8 @@ typedef enum {
   RET_HTTP_CERT,
   RET_HTTP_CONNECT,
   RET_HTTP_SSL,
-} http_retcode_t;
+  RET_HARDWARE_DEPS,
```
Comment (jserv) [inline utils/defined_error.h]: What is `RET_HARDWARE_DEPS`?

Comment (splasky) [reply utils/defined_error.h]: My failed.

Comment (jserv) [reply utils/defined_error.h]: Always write down meaningful comment and enumerations in header file.

**Comment by jserv** [comment]
You have to explain why and how you would like to perform these changes within both pull request and commit messages.

**Comment by howjmay** [comment]
@jserv This PR was originally open by me. I aimed to use a unique error return code throughout TA-endpoint

**Comment by jserv** [comment]
> @jserv This PR was originally open by me. I aimed to use a unique error return code throughout TA-endpoint

Make the task great again. Improve the descriptions in both issue tracker and pull request.

**Comment by howjmay** [comment]
> > @jserv This PR was originally open by me. I aimed to use a unique error return code throughout TA-endpoint
> 
> Make the task great again. Improve the descriptions in both issue tracker and pull request.

I will open a issue tracker for these.

**Review by howjmay — CHANGES_REQUESTED** [review]
Add `closes #37` in both git and PR messages.

**Review by howjmay — APPROVED** [review]

**Comment by jserv** [comment]
Use `git rebase -i` to rework commit messages. Less commits and more expressions.

**Review by jserv — CHANGES_REQUESTED** [review]
Improve the git commit message:
> The trytes message would be given by tangle accerlator, update the
comment.

Typo: "accerlator"
The grammar was poor.

**Comment by jserv** [comment]
Always say, "Tangle-accelerator." Be aware of the formal term.

**Comment by jserv** [comment]
There were some grammar errors in git commit messages. Use shell checkers to validate.

**Review by howjmay — CHANGES_REQUESTED** [review]
Fix gammar errors with the following setence.

1. `define_error.h` includes more general error codes. Therefore, we should change `http_retcode_t` to `retcode_t`.
2. Tangle-accelerator will provide trytes message, update the comment.

**Review by howjmay — CHANGES_REQUESTED** [review]
Fix test expression when `log_address()` returns. The return value should be non-zero if `log_address()` failed.

File: main.c [hunk 5]
```diff
@@ -99,8 +100,8 @@ int main(int argc, char *argv[]) {
     return -1;
   }
 #else
-  if (log_address(next_addr)) {
-    fprintf(stderr, "log address failed");
+  if (log_address(next_addr) != 0) {
```
Comment (jserv) [inline main.c]: Don't make unrelated changes.

File: main.c [hunk 6]
```diff
@@ -30,8 +30,8 @@
 
 static char addr_log_template[] = "\n%s\n";
 
+// Test mode only, should not use on deployment
```
Comment (jserv) [inline main.c]: This change was not related to the subject.

File: main.c [hunk 7]
```diff
@@ -100,7 +100,7 @@ int main(int argc, char *argv[]) {
   }
 #else
   if (log_address(next_addr)) {
-    fprintf(stderr, "log address failed");
+    fprintf(stderr, "log address failed\n");
```
Comment (jserv) [inline main.c]: Don't make any changes against `main.c`. Focus on your intended changes.

Comment (jserv) [reply main.c]: Never mix unrelated changes into single commit.

**Comment by jserv** [comment]
"Fix memory leak" should take place in another pull request.

**Comment by jserv** [comment]
Issue #37 explained the motivation and considerations for this change, and the commit message of this pull request and git should extend it as well.

File: main.c [hunk 8]
```diff
@@ -31,7 +31,6 @@
 static char addr_log_template[] = "\n%s\n";
 
 void gen_trytes(uint16_t len, char *out) {
-  // TODO: warning this may be risk
```
Comment (jserv) [inline main.c]: Don't touch this.

**Comment by jserv** [comment]
The message
> Renmae `http_retcode_t` to `retcode_t` for general error coverage.
lacks of informative reasons.
